scopes/compilation/compiler/workspace-compiler.ts

```diff
@@ -220,7 +220,9 @@ export class WorkspaceCompiler {
   ) {
     if (this.workspace) {
       this.workspace.registerOnComponentChange(this.onComponentChange.bind(this));
-      this.workspace.registerOnComponentAdd(this.onComponentChange.bind(this));
+      if (!this.dependencyResolver.hasRootComponents()) {
```

Comment:
add a comment here that if the root component is enabled we register from the installer which will call compile internally.